sdk/nodejs/api/api.ts

```diff
@@ -0,0 +1,447 @@
+   */
+  container(args?: QueryContainerArgs): Container {
+    this._queryTree = [
+      {
+      operation: 'container',
+      args
+      }
+    ]
+
+    return new Container(this._queryTree)
+  }
+  
+  /**
+   * Construct a cache volume for a given cache key
+   */
+  cacheVolume(args: {key: Scalars['String']}): CacheVolume {
+    this._queryTree = [
+      {
+      operation: 'cacheVolume',
+      args
+      }
+    ]
+
+    return new CacheVolume(this._queryTree)
+  }
+
+  /**
+   * Query a git repository
+   */
+  git(args: QueryGitArgs): Git {
+    this._queryTree = [
+      {
+      operation: 'git',
+      args
+      }
+    ]
+
+    return new Git(this._queryTree)
+  }
+
+  /**
+   * Query the host environment
+   */
+  host(): Host {
+    this._queryTree = [
+      {
+      operation: 'host',
+      }
+    ]
+
+    return new Host(this._queryTree)
+  }
+
+}
+
+class CacheVolume extends BaseClient {
+  /**
+   * A unique identifier for this container
+   */
+  async id(): Promise<Record<string, Scalars['CacheID']>> {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'id',
+      }
+    ]
+
+    const response: Awaited<Record<string, Scalars['CacheID']>> = await this._compute()
+
+    return response
+  }
+}
+
+class Host extends BaseClient {
+  /**
+   * The current working directory on the host
+   */
+  workdir(args?: HostWorkdirArgs): Directory {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'workdir',
+      args
+      }
+    ]
+
+    return new Directory(this._queryTree)
+  }
+}
+
+class Git extends BaseClient {
+  /**
+   * Details on one branch
+   */
+  branch(args: GitRepositoryBranchArgs): Tree {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+        operation: 'branch',
+        args
+      }
+    ]
+
+    return new Tree(this._queryTree)
+  }
+
+}
+class Tree extends BaseClient {
+  /**
+   * The filesystem tree at this ref
+   */
+  tree(): Directory {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+        operation: 'tree'
+      }
+    ]
+
+    return new Directory(this._queryTree)
+  }
+}
+class File extends BaseClient {
+    /**
+   * The contents of the file
+   */
+  async contents(): Promise<Record<string, string>> {
+    this._queryTree = [
+      ...this._queryTree, 
+      {
+      operation: 'contents'
+      }
+    ]
+
+    const response: Record<string, string> = await this._compute()
+
+    return response
+  }
+
+  /**
+   * The size of the file, in bytes
+   */
+  async size(): Promise<Record<string, number>> {
+    this._queryTree = [
+      ...this._queryTree, 
+      {
+      operation: 'size'
+      }
+    ]
+
+    const response: Record<string, number> = await this._compute()
+
+    return response
+  }
+}
+
+class Container extends BaseClient {
+
+  /**
+   * This container after executing the specified command inside it
+   */
+  exec(args: ContainerExecArgs): Container {
+    this._queryTree = [
```

Comment:
I think we want to leave `this._queryTree` alone and inline this in `return new Container` below

See `TestFieldImmutability` in the Go SDK.

If not, this will happen (pseudo code):

```go
container := c.Container().From("alpine")
a := container.Exec("echo", "hello", "world")
b := container.Exec("echo", "foo", "bar")
```

`b` will print both `hello world` and `foo bar` because `a := container.Exec()` modified the query of `container`

Reply:
If I try the scenario above:
```ts
const container = client.container().from({address: "alpine"})

const a = container.exec({args: ["echo", "hello", "world"]})
const b = await container.exec({args: ["echo", "foo", "bar"]}).stdout().contents()

// Print result
// { contents: 'foo bar' }
```

Reply:
@aluzzardi : Ok I get what you mean. The problem is I'll lose the context if I do not spread previous fields with the `...this.query_tree`.

Reply:
This has been fixed. Thanks @aluzzardi